batchflow/models/tf/mobilenet.py

```diff
@@ -219,8 +273,71 @@ def block(cls, inputs, filters, residual=False, strides=1, expansion_factor=6, w
         tf.Tensor
         """
         num_filters = int(cls.num_channels(inputs, kwargs.get('data_format')) * expansion_factor * width_factor)
-        conv_filters = [num_filters, num_filters, filters]
-        x = conv_block(inputs, 'cna Cna cn', conv_filters, [1, 3, 1], name=name, strides=[1, strides, 1], **kwargs)
+        x = conv_block(inputs, 'cnaWna', [num_filters, num_filters], [1, kernel_size], strides=[1, strides],
+                       name='%s-exp' % name, **kwargs)
+
+        if se_block:
+            if not isinstance(se_block, dict):
+                se_block = dict(activation=[kwargs.get('activation', tf.nn.relu), h_sigmoid], ratio=num_filters // 4)
+            se_block = {**kwargs, **se_block}
+            x = cls.se_block(x, name='%s-se' % name, **se_block)
+        x = conv_block(x, 'cn', filters, 1, name='%s-down' % name, **kwargs)
+
         if residual:
             x = inputs + x
         return x
+
+    @classmethod
+    def head(cls, inputs, se_block=None, name='head', **kwargs):
+        """ The last network layers which produce predictions
+
+        Parameters
+        ----------
+        inputs : tf.Tensor
+            input tensor
+        name : str
+            scope name
+        se_block : dict, optional
+            params for :meth:`~TFModel.se_block`
+
+        Returns
+        -------
+        tf.Tensor
+        """
+        kwargs = cls.fill_params('head', se_block=se_block, **kwargs)
+        layout, filters, se_block = cls.pop(['layout', 'filters', 'se_block'], kwargs)
+
+        if se_block:
+            x = conv_block(inputs, 'cna', filters[0], name='%s-conv1' % name, **kwargs)
+            x = cls.se_block(x, **{**kwargs, **se_block}, name='%s-se' % name)
+            x = conv_block(x, 'vcacV', filters=filters[1:], name='%s-conv2' % name, **kwargs)
+            return x
+        return conv_block(inputs, layout=layout, filters=filters, name=name, **kwargs)
+
+
+class MobileNet_v3(MobileNet_v2):
+    """ MobileNet version 3 large """
+    @classmethod
+    def default_config(cls):
+        config = TFModel.default_config()
+        config['initial_block'].update(dict(layout='cna', filters=16, kernel_size=3, strides=2, activation=h_swish))
+        config['body'].update(dict(width_factor=1, layout=deepcopy(_V3_LARGE_DEFAULT_BODY)))
```

Comment:
Looks like you need `deepcopy` because the exact keyword `layout` is used to store parameters, and you need to remove it in order to not interfere with the `layout` parameter of `conv_block`. Maybe it would be easier to change the word to, say, `architecture`, like it is done in `Inception` models?
Besides, there is a difference in formats of `_V1_DEFAULT_BODY ` and the others: the first contains `width_factor` in itself, while all of the latter make it separate parameter for the `body`

Reply:
Nope.